iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -3858,32 +3897,46 @@ STRING_HANDLE IoTHubTransport_MQTT_Common_GetHostname(TRANSPORT_LL_HANDLE handle
     return result;
 }
 
-IOTHUB_CLIENT_RESULT IoTHubTransport_MQTT_Common_SendMessageDisposition(MESSAGE_CALLBACK_INFO* message_data, IOTHUBMESSAGE_DISPOSITION_RESULT disposition)
+IOTHUB_CLIENT_RESULT IoTHubTransport_MQTT_Common_SendMessageDisposition(IOTHUB_DEVICE_HANDLE handle, IOTHUB_MESSAGE_HANDLE messageHandle, IOTHUBMESSAGE_DISPOSITION_RESULT disposition)
 {
-    (void)disposition;
-
     IOTHUB_CLIENT_RESULT result;
-    if (message_data)
+    if (handle == NULL || messageHandle == NULL)
     {
-        if (message_data->messageHandle)
-        {
-            IoTHubMessage_Destroy(message_data->messageHandle);
-            result = IOTHUB_CLIENT_OK;
-        }
-        else
-        {
-            /*Codes_SRS_IOTHUB_MQTT_TRANSPORT_10_002: [If any of the messageData fields are NULL, IoTHubTransport_MQTT_Common_SendMessageDisposition shall fail and return IOTHUB_CLIENT_ERROR. ]*/
-            LogError("message handle is NULL");
-            result = IOTHUB_CLIENT_ERROR;
-        }
-        free(message_data);
+        /*Codes_SRS_IOTHUB_MQTT_TRANSPORT_10_001: [If messageData is NULL, IoTHubTransport_MQTT_Common_SendMessageDisposition shall fail and return IOTHUB_CLIENT_ERROR. ]*/
+        LogError("Invalid argument (handle=%p, messageHandle=%p", handle, messageHandle);
+        result = IOTHUB_CLIENT_INVALID_ARG;
     }
     else
     {
-        /*Codes_SRS_IOTHUB_MQTT_TRANSPORT_10_001: [If messageData is NULL, IoTHubTransport_MQTT_Common_SendMessageDisposition shall fail and return IOTHUB_CLIENT_ERROR. ]*/
-        LogError("message_data is NULL");
-        result = IOTHUB_CLIENT_ERROR;
+        result = IOTHUB_CLIENT_OK;
```

Comment:
just double checking that the impl is correct here. If `disposition` is anything other than `IOTHUBMESSAGE_ACCEPTED`, then the message should be destroyed and the result should be `IOTHUB_CLIENT_OK`?

Reply:
yeah, that's how the upper layers are using the API, so since it's internal I kept it as is.

Reply:
I changed the code as per my last comment above. Thanks for initial poke, that brought my attention to that deficiency.